src/controllers/check.company.controller.ts

```diff
@@ -1,17 +1,45 @@
 import { NextFunction, Request, Response } from "express";
+import logger from "../logger";
+import { PTFCompanyProfile } from "../model/company.profile";
+import { getCompanyProfile } from "../client/apiclient";
+import * as errorMessages from "../model/error.messages";
+import * as templatePaths from "../model/template.paths";
+import * as sessionService from "../services/session.service";
 
 export const route = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
 
-    const scenario = { company: { PTFRequested: "Yes", accountsDue: "01 January 2020",
-            accountsStatus: "overdue", address: "25 No Street, Nowhere", companyType: "Private Limited Company",
-            csDue: "02 January 2020", csStatus: "overdue", incorporationDate: "01 January 2000",
-            name: "Test Company LTD", number: "00001111", status: "Active" }};
+    const companyNumber: string = sessionService.getCompanyInContext(req.chSession);
+    if (companyNumber) {
+        try {
+            logger.info(`Company number ${companyNumber} found in session, retrieving company profile`);
+            const token: string = req.chSession.accessToken() as string;
+            const companyinContext: PTFCompanyProfile = await getCompanyProfile(companyNumber, token);
+            logger.info("COMPANY PROFILE " + JSON.stringify(companyinContext));
+            const isDueDatePassed: boolean = checkDueDate(companyinContext);
+            return res.render(templatePaths.CHECK_COMPANY, {
+                    company: companyinContext,
+                    dueDatePassed: isDueDatePassed,
+                });
+        } catch (e) {
+            logger.error(`Error retrieving company number ${companyNumber} from redis`, e);
```

Comment:
Could this error message be mis-leading - what if there is actually a company number in redis but the call to the SDK on line 16 fails? This catch block would trigger (presumably) and message would be wrong. Perhaps just make the error message more generic.

Reply:
Shorted so that it doe snot mention redis just the error retrieving the company.